### ISSUE TITLE
longer description shouldn't be truncated to 100char limit

### ISSUE THREAD

**Issue by raibutera** [commitizen/cz-cli#4]
there's no point in having a 'longer description' if it is forced to be the same length as the short description.

**Comment by jimthedev** [comment]
Hi @raibutera. Thanks for the feedback. I'm assuming that you're referring to the body line mentioned here: https://github.com/ajoslin/conventional-changelog/blob/master/CONVENTIONS.md and that it is limited to 100 characters. 

As you can see, the conventional commit format dictates that any line should actually be limited to 100 characters, so we're just following the spec for now. You can remove this restriction by forking https://github.com/commitizen/cz-conventional-changelog and publishing your own adapter. You'd just edit the following line to remove the splice: https://github.com/commitizen/cz-conventional-changelog/blob/master/index.js#L80

I'm going to close this for now but if you feel we're misinterpreting the spec then please feel free to let me know.

**Comment by raibutera** [comment]
Jim, I do believe you are misinterpreting the specification.

"Any line of the commit message cannot be longer 100 characters! This
allows the message to be easier to read on github as well as in various git
tools."

The 100-character limit is PER LINE, not per SECTION. The body section can
comprise multiple LINES. In fact, it SHOULD (to preserve readability, as
per the justification of the limitation, ie. 'Allows the message to be
easier to read....').

tl;dr:

1) the 100 character limit is per line
2) the 100 character limit is for readability
3) applying the 100 character limit to each SECTION makes the body line
redundant (how can a more verbose body be achieved with the same characters
as the 'short' description?)
4) the point of the body section is to be verbose
5) look at angular commits, the body section is often longer than 100
characters.


Sorry for being blunt etc, I'm on mobile.

**Comment by jimthedev** [comment]
No problem about mobile bluntness. I understand. Sorry for my late reply. ;)  




You are indeed correct. So, rather than limiting the body field to 100 characters, we should do word wrapping for people at 100 characters but ultimately the field should be unlimited in length, correct? And the same would go for the footer which can be multiple lines and unlimited in length.




Does that sound right?




Jim

**Comment by jimthedev** [comment]
Quick update on this. I have a branch that works properly pending a [gulp-git pull request](https://github.com/stevelacy/gulp-git/pull/77) that includes some fixes for multi-line commits. Once that is release I'll release the new version of the cz-cli and cz-conventional-changelog adapter.

**Comment by jimthedev** [comment]
Please note that this is fixed in the adapter code in cz-conventional-changelog version 1.1.0 but in order to work properly it will still need a cli fix that is pending due to my gulp-git PR listed above.

**Comment by jimthedev** [comment]
This was released in [1.0.4](https://github.com/commitizen/cz-cli/releases/tag/1.0.4).
